File: 정수내림차순으로배치하기/javascript/20221019/solution.js
export function split(n) {
  return `${n}`.split('');
}

export function sort(array) {
  return [...array].map(Number).sort((a, b) => b - a);
}

export function join(array) {
  return parseInt(array.join(''), 10);
}

function solution(n) {
  const sortedNumbers = sort(split(n));
  return join(sortedNumbers);
}

File: 정수내림차순으로배치하기/java/20221019/src/Solution.java
import java.util.Arrays;
import java.util.Comparator;

public class Solution {
    public long solution(long n) {
        String[] sortedNumbers = sort(split(n));
        return joinToLong(sortedNumbers);
    }

    public String[] split(long n) {
        String number = String.valueOf(n);

        return number.split("");
    }

    public String[] sort(String[] numbers) {
        return Arrays.stream(numbers)
                .sorted(Comparator.reverseOrder())
                .toArray(String[]::new);
    }

    public long joinToLong(String[] sortedNumbers) {
        String joinedNumbers = String.join("", sortedNumbers);
        return Long.parseLong(joinedNumbers);
    }
}
